File: 682-baseball-game/baseball-game.py
class Solution:
    def calPoints(self, operations: List[str]) -> int:
        stack = []
        
        for op in operations:
            if op == '+':
                # Sum of the previous two scores
                stack.append(stack[-1] + stack[-2])
            elif op == 'D':
                # Double of the previous score
                stack.append(2 * stack[-1])
            elif op == 'C':
                # Invalidate and remove the previous score
                stack.pop()
            else:
                # It's an integer score
                stack.append(int(op))
        
        return sum(stack)

File: 682-baseball-game/baseball-game.cpp
class Solution {
public:
    int calPoints(vector<string>& operations) {
        vector<int> stack;
        int total = 0;
        
        for (const string& op : operations) {
            if (op == "+") {
                int n = stack.size();
                int val = stack[n - 1] + stack[n - 2];
                stack.push_back(val);
                total += val;
            } else if (op == "D") {
                int val = 2 * stack.back();
                stack.push_back(val);
                total += val;
            } else if (op == "C") {
                total -= stack.back();
                stack.pop_back();
            } else {
                int val = stoi(op);
                stack.push_back(val);
                total += val;
            }
        }
        
        return total;
    }
};